src/types/list.rs

```diff
@@ -83,17 +83,34 @@ impl PyList {
         }
     }
 
+    /// Takes the slice `self[low:high]` and returns it as a new list.
+    ///
+    /// Out-of-range indices are clipped to `self.len()`.
+    pub fn slice(&self, low: usize, high: usize) -> &PyList {
+        unsafe {
+            self.py().from_owned_ptr(ffi::PyList_GetSlice(
+                self.as_ptr(),
+                low as Py_ssize_t,
+                high as Py_ssize_t,
```

Comment:
Because this is an "as" cast, there's actually something weird here where users can (ab)use twos complement to smuggle negative indices into this call: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=ec2001b00be41c47404ab07d3826633c

```
[src/main.rs:2] std::usize::MAX as isize = -1
```

We might need to consider overflow behaviour like this for all of the functions in this file?

For `get_slice`, it given that out-of-range indices are already clipped at the top end, we may wish to just clamp values to std::isize::MAX before converting? E.g.

```suggestion
                std::cmp::min(low, std::isize::MAX as usize) as Py_ssize_t,
                std::cmp::min(high, std::isize::MAX as usize) as Py_ssize_t,
```